Project lifecycle docs (#40)
sonatype-nexus-community/contribute.sonatype.com: content/en/docs/project-lifecycle/activation.md

---
title: Activating a Project
weight: 70
---

Projects within our Community may occasionally transition between states, such as moving from an unset or archived status to active. This process ensures projects are thoughtfully evaluated and meet the standards required for active maintenance.

## Activating a Project

To activate a project, follow these steps:

### 1. Review the Project

Assess the project's relevance, dependencies, and community interest. This includes:

- Consider recent issues, pull requests, or community engagement
- Review alignment with current Community goals, as applicable
- Confirm the project’s compliance with Community [Project Standards](https://contribute.sonatype.com/docs/standards/)

### 2. Assign a Maintainer

> Ensure a maintainer is identified to oversee the project and handle ongoing contributions. This step is crucial for maintaining quality and responsiveness.

Reviewer: ```suggestion
Ensure at least one maintainer is identified to oversee the project and handle ongoing contributions. This step is crucial for maintaining quality and responsiveness.
```

> Ensure at least one maintainer is identified to oversee the project and handle ongoing contributions. This step is crucial for maintaining quality and responsiveness.

Reviewer 2: Perhaps a link to our docs about "Maintainer" here? (Also not sure if as a general rule we want to link to the .md files, or the published pages? like: https://contribute.sonatype.com/docs/community_roles/maintainer/ vs ./community_roles/maintainer.md)

```suggestion
Ensure at least one [maintainer](https://contribute.sonatype.com/docs/community_roles/maintainer/) is identified to oversee the project and handle ongoing contributions. This step is crucial for maintaining quality and responsiveness.
```

> Maintainers MUST have previously contributed to another Sonatype Open Source Project. This ensures they are familiar with our Community workflows and have signed our CLA.

Reviewer 2: I'm hoping this requirement does not become a road block. Not concerned enough to remove it though. Maybe it could be relaxed to it is "Preferred maintainers have previously contributed...."

> Assigning the initial maintainer for a project requires unanimous approval of the [Community Admins](../community_roles/admin.md). 

Reviewer: ```suggestion
Assigning the initial maintainers for a project requires unanimous approval of the [Community Admins](../community_roles/admin.md). 
```

{{< alert title="Bring your friends" >}}
Adding additional maintainers may be added at this stage OR a later date with unanimous approval from the existing maintainers and support from at least one Community Admin.
{{< /alert >}}

### 3. Update the Custom Property

Modify the **Project-Status** property using [GitHub Custom Properties](https://docs.github.com/en/organizations/managing-organization-settings/managing-custom-properties-for-repositories-in-your-organization):
- **Set to `Active`** to indicate the project is actively maintained.

For projects previously archived:
- First, unarchive the repository via the GitHub interface.

### 4. Communicate the Change

- Announce the project’s activation in relevant Community channels.
- Update any associated documentation to reflect its active status.

## Qualifying Projects

> For a list of projects currently archived or unset, use the following links:

Reviewer: What is unset?

Author: The "unset" phrasing comes from here:

https://github.com/sonatype-nexus-community/contribute.sonatype.com/blob/main/content/en/docs/project-classification/_index.md

Probably could use a bit of language refinement here 🤔
